cmd/okctl/scaffoldapplication.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+
+	kaex "github.com/oslokommune/kaex/pkg/api"
+	"github.com/oslokommune/okctl/pkg/scaffold"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/oslokommune/okctl/pkg/okctl"
+	"github.com/spf13/cobra"
+)
+
+const requiredArgumentsForCreateApplicationCommand = 1
+
+// CreateApplicationOpts contains all the possible options for "create application"
+type CreateApplicationOpts struct {
+	Environment string
+	KaexOpts    *kaex.Kaex
+}
+
+// Validate the options for "create application"
+func (o *CreateApplicationOpts) Validate() error {
+	return validation.ValidateStruct(o,
+		validation.Field(&o.Environment),
+	)
+}
+
+func buildCreateApplicationCommand(o *okctl.Okctl) *cobra.Command {
+	opts := &CreateApplicationOpts{}
+	interpolationOpts := &scaffold.InterpolationOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "application ENV",
+		Short: "Scaffold an application template",
+		Long:  "Scaffolds an application.yaml template which can be used to produce necessary Kubernetes and ArgoCD resources",
+		Args:  cobra.ExactArgs(requiredArgumentsForCreateApplicationCommand),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.Environment = args[0]
+
+			err := o.InitialiseWithOnlyEnv(opts.Environment)
+			if err != nil {
+				return err
+			}
+
+			hostedZone := o.RepoStateWithEnv.GetPrimaryHostedZone()
+			interpolationOpts.Domain = hostedZone.Domain
+
+			opts.KaexOpts = &kaex.Kaex{
+				Err:             o.Err,
+				Out:             o.Out,
+				In:              o.In,
+				TemplatesDirURL: "https://raw.githubusercontent.com/oslokommune/kaex/master/templates",
+			}
+
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := opts.Validate()
+			if err != nil {
+				return err
+			}
+
+			template, err := scaffold.FetchTemplate(*opts.KaexOpts)
+			if err != nil {
+				return fmt.Errorf("failed fetching application.yaml example: %w", err)
+			}
+
+			interpolatedResult, err := scaffold.InterpolateTemplate(template, interpolationOpts)
+			if err != nil {
+				return err
+			}
+
+			err = scaffold.SaveTemplate(interpolatedResult)
+			if err != nil {
+				return err
+			}
+
+			fmt.Fprintln(o.Out, "Scaffolding successful.")
+			fmt.Fprintf(o.Out, "Edit ./application.yaml to your liking and run okctl apply application %s -f application.yaml\n", opts.Environment)
```

Comment:
Just a check: Is application still in the current directory?

Reply:
Hmm. This is outdated. Directory is by default the current directory, but can now be specified with --out

Reply:
Yes, and if somebody specify -out, wouldn't this output to the user be invalid? It should be "Edit %s/application.yaml [...]" where %s is the directory specified in --out ?

Reply:
@yngvark the code you pointed to here is outdated.

Relevant code is [here](https://github.com/oslokommune/okctl/pull/264/commits/e1cbe55ec7a0040ff0ed4859543b4ec8ea7c7cae#diff-54bdf823ee965e370ebcfa021a42dba2a92377344927dd2378acc77dfa6bb46cR84) where I've changed the output